vector-stores/spring-ai-azure-store/src/main/java/org/springframework/ai/vectorstore/azure/AzureVectorStore.java

```diff
@@ -379,4 +406,13 @@ public void afterPropertiesSet() throws Exception {
 		this.searchClient = this.searchIndexClient.getSearchClient(this.indexName);
 	}
 
+	@Override
+	public Builder createObservationContextBuilder(String operationName) {
+
+		return VectorStoreObservationContext.builder(VectorStoreProvider.AZURE_VECTOR_STORE.value(), operationName)
+			.withDimensions(this.embeddingModel.dimensions())
+			.withSimilarityMetric(this.initializeSchema ? VectorStoreSimilarityMetric.COSINE.value() : "Unknown")
```

Comment:
Should it be `null` instead of `"Unknown"`, so that the observation value will end up being `KeyValue.NONE`? For other attributes, when we can't infer the value, we fall back to "none".